### DISCUSSION TITLE
Changing the `id` field type , when I have not manually specified in the table, fails in migrations

### DISCUSSION THREAD

**Discussion by sumitsharansatsangi** [piccolo-orm/piccolo#765] [Ideas]
Earlier I have table as follows:
```python
class Band(Table):
    name = Varchar(length=100)
```
When I am changing it to following , migration is giving the error that id already exists.
```python
class Band(Table):
    id = UUID(primary_key=True)
    name = Varchar(length=100)
```

The problem is piccolo is considering the `id ` field as it is newly added.
In the migration file , piccolo is generating the migration for `add_column` instead of `alter_column`

One way to solve this problem is that if the table already exist and the column which is added is a primary key then it should generate the migration of alter_column, because piccolo auto generates only the id column as primary key , if it is not manually specified in the table.

Above suggested way will work in the following condition too( Where a different name is specified for the primary key), with a modification which is that `piccolo` has to generate migration for `rename_column` too, ( for now without asking a question `Did you rename the column in the table ....` till piccolo doesn't support for complex primary key and specifying a complex primary key is not like as follows
```python
class Band(Table):
    band_id = UUID(primary_key=True)
    name = Varchar(length=100)
    type = ForeignKey(references = BandType ,primary_key=True)
```
)

Before:
```python
class Band(Table):
    name = Varchar(length=100)
```

After:
```python
class Band(Table):
    band_id = UUID(primary_key=True)
    name = Varchar(length=100)
```

**Comment by dantownsend** [comment]
Yeah, it's tricky for Piccolo to change the primary key of a table, because it would also have to update any foreign keys pointing to the table.

I agree we could do something better though.

If the table has no data, one solution is to drop the table (just comment it out, and create a new migration). Then re-add it with the correct foreign key.

**Reply by sumitsharansatsangi** [reply]
Yeah, I have done the same. You are right, Updating Foreign key may be problematic.
We have to look into the implementation of some mature orm.
